File: contains-duplicate-121/contains_duplicate.go
package main

func containsDuplicate(nums []int) bool {
    is_seen := make(map[int]int)
    for num:= range nums {
        _, ok := is_seen[num]
        if ok {
            return true
        }

        is_seen[num] = 1

    }

    return false
}

func main() {
   containsDuplicate([]int{1,2,3,1})
}

File: contains-duplicate-121/contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int,int> is_seen;
        for(int i=0; i< nums.size(); i++) {
            if(is_seen.count(nums[i]) > 0) {
                return true;
            }
            is_seen[nums[i]] = 1;
        }
        return false;
    }
};